prac_05/emails.py

```diff
@@ -0,0 +1,29 @@
+"""
+CP1404/CP5632 Practical
+Email and name extractor program
+"""
+
+
+def main():
+    """Create a dictionary of emails to names."""
+    email_to_name = {}
+    email = input("Email: ")
+    while email != "":
+        name = extract_name(email)
+        email_to_name[email] = name
+        email = input("Email: ")
+    for email, name in email_to_name.items():
+        print(f"{name} ({email})")
+
+
+def extract_name(email):
+    """Extract the name from a provided email address."""
```

Comment:
Lindsays solution has the confirmation of name in the main function.
Your placement makes sense to me, maybe add to the docstring that this function also confirms name.